lib/utils.js

```
@@ -1,18 +1,34 @@
     return ctx ? fn.call(ctx, ...args) : fn(...args);
   },
 
   middleware(fn) {
-    return is.generatorFunction(fn) ? convert(fn) : fn;
+    return isGeneratorFunction(fn) ? convert(fn) : fn;
+  },
+
+  isGeneratorFunction,
+  isAsyncFunction(obj) {
+    return obj
+      && obj.constructor
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)